sdk/eventhub/Azure.Messaging.EventHubs/src/Core/TransportProducerPool.cs

```diff
@@ -165,7 +173,7 @@ internal class TransportProducerPool : IAsyncDisposable
         ///
         /// <returns>A task to be resolved on when the operation has completed.</returns>
         ///
-        public async Task CloseAsync(CancellationToken cancellationToken = default)
+        public virtual async Task CloseAsync(CancellationToken cancellationToken = default)
         {
             await EventHubProducer.CloseAsync(true, cancellationToken).ConfigureAwait(false);
```

Comment:
You're not going to see one producer interfere with the others during close, at least outside the `EventHubProducer` which we should add to the pending closes.    The `Task.WhenAll` will make sure to allow all tasks to complete before throwing any failures.

Reply:
I moved ```ExpirationTimer.Dispose();``` at the start of ```CloseAsync```.